yt/data_objects/static_output.py

```diff
@@ -223,18 +223,7 @@ def __init__(
         self.units_override = self.__class__._sanitize_units_override(units_override)
         self.default_species_fields = default_species_fields
 
-        # path stuff
-        filename = os.path.expanduser(filename)
-        self.parameter_filename = filename
-        self.basename = os.path.basename(filename)
-        self.directory = os.path.dirname(filename)
-        self.fullpath = os.path.abspath(self.directory)
-        self.backup_filename = self.parameter_filename + "_backup.gdf"
-        self.read_from_backup = False
-        if os.path.exists(self.backup_filename):
-            self.read_from_backup = True
-        if len(self.directory) == 0:
-            self.directory = "."
+        self._input_filename = os.fspath(filename)
```

Comment:
os.fspath is great here because it invalidates a `None` and many objects that can't represent paths.
It also normalises `pathlib.Path` objects to `str`, which is desirable. However it's not sufficient to invalidate `bytes` -typed inputs (considered valid representations of paths, but incompatible with this attribute in many places where we implicitly assume a `str`).